### PR TITLE
[1116][ADD] hide_partner_without_moves filter to general ledger

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
Task #[1116](https://www.quartile.co/web#id=1116&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

File: account_financial_report/README.rst [hunk 1]
```diff
@@ -42,6 +42,8 @@ currency used in account move lines is properly shown.
 In case that in an account has not been configured a second currency foreign
 currency balances are not available.
 
+Added `Hide Partners without Moves` filter: This filter hides the moves where they don't have `partner_id`
```
Comment (yostashiro) [inline account_financial_report/README.rst]: @YogeshMahera-SerpentCS I think there is misunderstanding on the requirement. The requirement is to not show partners with no moves, which is different to hiding moves with no partner.

Comment (YogeshMahera-SerpentCS) [reply account_financial_report/README.rst]: Peek 2020-08-05 22-34

Hi, @yostashiro I have added sample GIF and Can you please check above sample GIF?

Comment (yostashiro) [reply account_financial_report/README.rst]: @YogeshMahera-SerpentCS I can confirm that the requirement has been misunderstood.

File: account_financial_report_extended/wizard/general_ledger_wizard.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    def _prepare_report_general_ledger(self):
+        res = super(GeneralLedgerReportWizard, self)._prepare_report_general_ledger()
+        if self.hide_partner_without_moves and not self.partner_ids:
+            domain = [
+                ("date", ">=", self.date_from),
+                ("date", "<=", self.date_to),
+                ("company_id", "=", self.company_id.id),
+            ]
+            if self.target_move == "posted":
+                domain += [("state", "=", self.target_move)]
```
Comment (yostashiro) [inline account_financial_report_extended/wizard/general_ledger_wizard.py]: We should consider all the relevant parameters from here https://github.com/qrtl/asx-custom/blob/2011c132fef6b20ed497382e92558de4cf1753f2/account_financial_report/wizard/general_ledger_wizard.py#L235-L252, not only the date range and the company.

File: account_financial_report_extended/wizard/general_ledger_wizard.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            account_move = self.env["account.move"].search(domain)
+            partner_ids = account_move.mapped("line_ids").mapped("partner_id")
+            res.update({"filter_partner_ids": [(6, 0, partner_ids.ids)]})
```
Comment (yostashiro) [inline account_financial_report_extended/wizard/general_ledger_wizard.py]: We should not simply replace the partner ids here, disregarding what the user has set in the wizard.

**Review by yostashiro — CHANGES_REQUESTED** [review]
@YogeshMahera-SerpentCS Can you check if following changes will do?

```.py
class GeneralLedgerReportWizard(models.TransientModel):
    _inherit = "general.ledger.report.wizard"

    hide_partner_without_moves = fields.Boolean(
        string="Hide Partners without Moves",
    )

    def _prepare_report_general_ledger(self):
        res = super()._prepare_report_general_ledger()
        res["hide_partner_without_moves"] = self.hide_partner_without_moves 
        return res
```
```.py
class GeneralLedgerReportCompute(models.TransientModel):
    _inherit = "report_general_ledger"

    @api.multi
    def compute_data_for_report(self,
                                with_line_details=True,
                                with_partners=True):
        super().compute_data_for_report(self, with_line_details, with_partners)
        if self.hide_partner_without_moves:
          self.env["report_general_ledger_partner"].search([("move_line_ids", "=", False)]).unlink
```

**Comment by YogeshMahera-SerpentCS** [comment]
Hi, @yostashiro I have tried your suggestion: It seems to returns the same output, which I have written the logic for partner_ids. also the your logic is much better.

```
class GeneralLedgerReportWizard(models.TransientModel):
    _inherit = "general.ledger.report.wizard"

    hide_partner_without_moves = fields.Boolean(
        string="Hide Partners without Moves",
    )

    def _prepare_report_general_ledger(self):
        res = super()._prepare_report_general_ledger()
        res["hide_partner_without_moves"] = self.hide_partner_without_moves
        return res

class GeneralLedgerReportCompute(models.TransientModel):
    _inherit = "report_general_ledger"

    @api.multi
    def compute_data_for_report(self,
                                with_line_details=True,
                                with_partners=True):
        super().compute_data_for_report(with_line_details, with_partners)
        if self.hide_partner_without_moves:
            self.env["report_general_ledger_partner"].search(
                [("move_line_ids", "=", False)]).unlink()

```

File: account_financial_report_extended/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+This module extend the General ledger function from financial report.
+ - Adds `Hide Partners without Moves` filter
```
Comment (yostashiro) [inline account_financial_report_extended/readme/DESCRIPTION.rst]:
```suggestion
This module extends the General Ledger function from financial reports.
 - Adds 'Hide Partners without Moves' filter
```

File: account_financial_report_extended/wizard/general_ledger_wizard.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+from odoo import fields, models
+
+
+class GeneralLedgerReportWizard(models.TransientModel):
+    _inherit = "general.ledger.report.wizard"
+
+    hide_partner_without_moves = fields.Boolean(
+        string="Hide Partners without Moves",
+        help="Hide the Accounts who don't have the partners",
```
Comment (yostashiro) [inline account_financial_report_extended/wizard/general_ledger_wizard.py]:
```suggestion
        help="If selected, partners that do not have account moves will not show in the result for payable and receivable accounts.",
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
Please update the description/help.  LGTM, otherwise.

**Comment by YogeshMahera-SerpentCS** [comment]
Hi @TimLai125 can you please review it?

File: sale_invoice_enduser/tests/__init__.py [hunk 6]
```diff
@@ -0,0 +1 @@
+from . import test_sale_order
```
Comment (yostashiro) [inline sale_invoice_enduser/tests/__init__.py]: @YogeshMahera-SerpentCS Please do not include irrelevant update here. Why don't you double check the entire content of PR yourself (this kind of mistake should be easy to spot) before asking for a review?

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: sale_invoice_enduser/tests/__init__.py (hunk 6 above)
Comment (YogeshMahera-SerpentCS) [reply sale_invoice_enduser/tests/__init__.py]: Thanks, @yostashiro for the check.

**Comment by YogeshMahera-SerpentCS** [comment]
Hi @yostashiro @TimLai125 Can you please review it?

**Review by yostashiro — APPROVED** [review]

**Pull request merged**